src/NeuralNetwork/index.js

```diff
@@ -500,7 +478,34 @@ class NeuralNetwork {
    * @param {*} sample 
    */
   async predictInternal(sample) {
-    const xs = tf.tensor(sample, [1, sample.length]);
+
+    // Handle the input sample
+    // either an array of values in order of the inputs
+    // OR an JSON object of key/values
+
+    let inputData = [];
+    if (sample instanceof Array) {
+
+      inputData = sample;
+
+    } else if (sample instanceof Object) {
+
+      // TODO: check if the input order is preserved!
+      const headers = Object.keys(this.data.inputs);
```

Comment:
I have been poking around and I might not be understanding this correctly, but I believe this is an error since `this.data.inputs` is an array already, not an Object with keys.
```suggestion
      const headers = this.data.inputs;
```

Reply:
Oh yes, this should be fixed now with the latest.